File: VB/WindowsApplication1/TreeListFormatConditions.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports System.Collections
Imports DevExpress.XtraTreeList.StyleFormatConditions
Imports DevExpress.Utils
Imports DevExpress.XtraGrid
Imports DevExpress.Utils.Serializing

Namespace WindowsApplication1
	Public Class TreeListFormatConditions

		Private _conditions As SerializableFormatCollection

		Public Sub New()
			_conditions = New SerializableFormatCollection()
		End Sub

		Public Sub New(ByVal styles As StyleFormatConditionCollection)
			_conditions = New SerializableFormatCollection(styles)
		End Sub

		<XtraSerializableProperty(XtraSerializationVisibility.Collection, True, False, True, 0, XtraSerializationFlags.None)> _
		Public ReadOnly Property Conditions() As SerializableFormatCollection
			Get
				Return _conditions
			End Get
		End Property

		Private Function XtraCreateConditionsItem(ByVal e As XtraItemEventArgs) As Object
			Return Conditions.Add()
		End Function
	End Class
End Namespace

File: CS/WindowsApplication1/TreeListFormatConditions.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Collections;
using DevExpress.XtraTreeList.StyleFormatConditions;
using DevExpress.Utils;
using DevExpress.XtraGrid;
using DevExpress.Utils.Serializing;

namespace WindowsApplication1
{
    public class TreeListFormatConditions
    {

        private SerializableFormatCollection _conditions;

        public TreeListFormatConditions()
        {
            _conditions = new SerializableFormatCollection();
        }

        public TreeListFormatConditions(StyleFormatConditionCollection styles)
        {
            _conditions = new SerializableFormatCollection(styles);
        }

        [XtraSerializableProperty(XtraSerializationVisibility.Collection, true, false, true, 0, XtraSerializationFlags.None)]
        public SerializableFormatCollection Conditions
        {
            get { return _conditions; }
        }

        private object XtraCreateConditionsItem(XtraItemEventArgs e)
        {
            return Conditions.Add();
        }
    }
}
